pkg/chunkenc/unordered.go

```diff
@@ -38,10 +68,29 @@ func newUnorderedHeadBlock() *unorderedHeadBlock {
 	}
 }
 
-func (hb *unorderedHeadBlock) isEmpty() bool {
+func (hb *unorderedHeadBlock) Format() HeadBlockFmt { return UnorderedHeadBlockFmt }
+
+func (hb *unorderedHeadBlock) IsEmpty() bool {
 	return hb.size == 0
 }
 
+func (hb *unorderedHeadBlock) Bounds() (int64, int64) {
+	return hb.mint, hb.maxt
+}
+
+func (hb *unorderedHeadBlock) Entries() int {
+	return hb.lines
+}
+
+func (hb *unorderedHeadBlock) UncompressedSize() int {
+	return hb.size
+}
+
+func (hb *unorderedHeadBlock) Reset() {
+	x := newUnorderedHeadBlock()
```

Comment:
Might be interesting instead to use https://github.com/golang-collections/go-datastructures/blob/59788d5eb259/rangetree/entries.go#L33 which will attemps to reuse entries.

Reply:
ooh great catch